userspace/libsinsp/metrics_collector.cpp

```diff
@@ -0,0 +1,559 @@
+			m_metrics.emplace_back(new_metric(sinsp_stats_v2_resource_utilization_names[SINSP_STATS_V2_STORE_EVTS_DROPS], \
+			METRICS_V2_STATE_COUNTERS, METRIC_VALUE_TYPE_U64, METRIC_VALUE_UNIT_COUNT, METRIC_VALUE_MONOTONIC, sinsp_stats_v2->m_n_store_evts_drops));
+			m_metrics.emplace_back(new_metric(sinsp_stats_v2_resource_utilization_names[SINSP_STATS_V2_RETRIEVED_EVTS], \
+			METRICS_V2_STATE_COUNTERS, METRIC_VALUE_TYPE_U64, METRIC_VALUE_UNIT_COUNT, METRIC_VALUE_MONOTONIC, sinsp_stats_v2->m_n_retrieved_evts));
+			m_metrics.emplace_back(new_metric(sinsp_stats_v2_resource_utilization_names[SINSP_STATS_V2_RETRIEVE_EVTS_DROPS], \
+			METRICS_V2_STATE_COUNTERS, METRIC_VALUE_TYPE_U64, METRIC_VALUE_UNIT_COUNT, METRIC_VALUE_MONOTONIC, sinsp_stats_v2->m_n_retrieve_evts_drops));
+			m_metrics.emplace_back(new_metric(sinsp_stats_v2_resource_utilization_names[SINSP_STATS_V2_NONCACHED_THREAD_LOOKUPS], \
+			METRICS_V2_STATE_COUNTERS, METRIC_VALUE_TYPE_U64, METRIC_VALUE_UNIT_COUNT, METRIC_VALUE_MONOTONIC, sinsp_stats_v2->m_n_noncached_thread_lookups));
+			m_metrics.emplace_back(new_metric(sinsp_stats_v2_resource_utilization_names[SINSP_STATS_V2_CACHED_THREAD_LOOKUPS], \
+			METRICS_V2_STATE_COUNTERS, METRIC_VALUE_TYPE_U64, METRIC_VALUE_UNIT_COUNT, METRIC_VALUE_MONOTONIC, sinsp_stats_v2->m_n_cached_thread_lookups));
+			m_metrics.emplace_back(new_metric(sinsp_stats_v2_resource_utilization_names[SINSP_STATS_V2_FAILED_THREAD_LOOKUPS], \
+			METRICS_V2_STATE_COUNTERS, METRIC_VALUE_TYPE_U64, METRIC_VALUE_UNIT_COUNT, METRIC_VALUE_MONOTONIC, sinsp_stats_v2->m_n_failed_thread_lookups));
+			m_metrics.emplace_back(new_metric(sinsp_stats_v2_resource_utilization_names[SINSP_STATS_V2_ADDED_THREADS], \
+			METRICS_V2_STATE_COUNTERS, METRIC_VALUE_TYPE_U64, METRIC_VALUE_UNIT_COUNT, METRIC_VALUE_MONOTONIC, sinsp_stats_v2->m_n_added_threads));
+			m_metrics.emplace_back(new_metric(sinsp_stats_v2_resource_utilization_names[SINSP_STATS_V2_REMOVED_THREADS], \
+			METRICS_V2_STATE_COUNTERS, METRIC_VALUE_TYPE_U64, METRIC_VALUE_UNIT_COUNT, METRIC_VALUE_MONOTONIC, sinsp_stats_v2->m_n_removed_threads));
+			m_metrics.emplace_back(new_metric(sinsp_stats_v2_resource_utilization_names[SINSP_STATS_V2_N_DROPS_FULL_THREADTABLE], \
+			METRICS_V2_STATE_COUNTERS, METRIC_VALUE_TYPE_U32, METRIC_VALUE_UNIT_COUNT, METRIC_VALUE_MONOTONIC, sinsp_stats_v2->m_n_drops_full_threadtable));
+			m_metrics.emplace_back(new_metric(sinsp_stats_v2_resource_utilization_names[SINSP_STATS_V2_N_MISSING_CONTAINER_IMAGES], \
+			METRICS_V2_STATE_COUNTERS, METRIC_VALUE_TYPE_U32, METRIC_VALUE_UNIT_COUNT, METRIC_VALUE_NON_MONOTONIC_CURRENT, sinsp_stats_v2->m_n_missing_container_images));
+			m_metrics.emplace_back(new_metric(sinsp_stats_v2_resource_utilization_names[SINSP_STATS_V2_N_CONTAINERS], \
+			METRICS_V2_STATE_COUNTERS, METRIC_VALUE_TYPE_U32, METRIC_VALUE_UNIT_COUNT, METRIC_VALUE_NON_MONOTONIC_CURRENT, sinsp_stats_v2->m_n_containers));
+		}
```

Comment:
Same spacing suggestion
```suggestion
			m_metrics.emplace_back(new_metric(sinsp_stats_v2_resource_utilization_names[SINSP_STATS_V2_N_THREADS],
			                                    METRICS_V2_STATE_COUNTERS,
                                                METRIC_VALUE_TYPE_U64,
                                                METRIC_VALUE_UNIT_COUNT,
                                                METRIC_VALUE_NON_MONOTONIC_CURRENT,
                                                n_threads));
			m_metrics.emplace_back(new_metric(sinsp_stats_v2_resource_utilization_names[SINSP_STATS_V2_N_FDS],
			                                    METRICS_V2_STATE_COUNTERS,
                                                METRIC_VALUE_TYPE_U64,
                                                METRIC_VALUE_UNIT_COUNT,
                                                METRIC_VALUE_NON_MONOTONIC_CURRENT,
                                                n_fds));
			m_metrics.emplace_back(new_metric(sinsp_stats_v2_resource_utilization_names[SINSP_STATS_V2_NONCACHED_FD_LOOKUPS], \
			                                    METRICS_V2_STATE_COUNTERS,
                                                METRIC_VALUE_TYPE_U64,
                                                METRIC_VALUE_UNIT_COUNT,
                                                METRIC_VALUE_MONOTONIC,
                                                sinsp_stats_v2->m_n_noncached_fd_lookups));
			m_metrics.emplace_back(new_metric(sinsp_stats_v2_resource_utilization_names[SINSP_STATS_V2_CACHED_FD_LOOKUPS], \
			                                    METRICS_V2_STATE_COUNTERS,
                                                METRIC_VALUE_TYPE_U64,
                                                METRIC_VALUE_UNIT_COUNT,
                                                METRIC_VALUE_MONOTONIC,
                                                sinsp_stats_v2->m_n_cached_fd_lookups));
			m_metrics.emplace_back(new_metric(sinsp_stats_v2_resource_utilization_names[SINSP_STATS_V2_FAILED_FD_LOOKUPS], \
			                                    METRICS_V2_STATE_COUNTERS,
                                                METRIC_VALUE_TYPE_U64,
                                                METRIC_VALUE_UNIT_COUNT,
                                                METRIC_VALUE_MONOTONIC,
                                                sinsp_stats_v2->m_n_failed_fd_lookups));
			m_metrics.emplace_back(new_metric(sinsp_stats_v2_resource_utilization_names[SINSP_STATS_V2_ADDED_FDS], \
			                                    METRICS_V2_STATE_COUNTERS,
                                                METRIC_VALUE_TYPE_U64,
                                                METRIC_VALUE_UNIT_COUNT,
                                                METRIC_VALUE_MONOTONIC,
                                                sinsp_stats_v2->m_n_added_fds));
			m_metrics.emplace_back(new_metric(sinsp_stats_v2_resource_utilization_names[SINSP_STATS_V2_REMOVED_FDS], \
			                                    METRICS_V2_STATE_COUNTERS,
                                                METRIC_VALUE_TYPE_U64,
                                                METRIC_VALUE_UNIT_COUNT,
                                                METRIC_VALUE_MONOTONIC,
                                                sinsp_stats_v2->m_n_removed_fds));
			m_metrics.emplace_back(new_metric(sinsp_stats_v2_resource_utilization_names[SINSP_STATS_V2_STORED_EVTS], \
			                                    METRICS_V2_STATE_COUNTERS,
                                                METRIC_VALUE_TYPE_U64,
                                                METRIC_VALUE_UNIT_COUNT,
                                                METRIC_VALUE_MONOTONIC,
                                                sinsp_stats_v2->m_n_stored_evts));
			m_metrics.emplace_back(new_metric(sinsp_stats_v2_resource_utilization_names[SINSP_STATS_V2_STORE_EVTS_DROPS], \
			                                    METRICS_V2_STATE_COUNTERS,
                                                METRIC_VALUE_TYPE_U64,
                                                METRIC_VALUE_UNIT_COUNT,
                                                METRIC_VALUE_MONOTONIC,
                                                sinsp_stats_v2->m_n_store_evts_drops));
			m_metrics.emplace_back(new_metric(sinsp_stats_v2_resource_utilization_names[SINSP_STATS_V2_RETRIEVED_EVTS], \
			                                    METRICS_V2_STATE_COUNTERS,
                                                METRIC_VALUE_TYPE_U64,
                                                METRIC_VALUE_UNIT_COUNT,
                                                METRIC_VALUE_MONOTONIC,
                                                sinsp_stats_v2->m_n_retrieved_evts));
			m_metrics.emplace_back(new_metric(sinsp_stats_v2_resource_utilization_names[SINSP_STATS_V2_RETRIEVE_EVTS_DROPS], \
			                                    METRICS_V2_STATE_COUNTERS,
                                                METRIC_VALUE_TYPE_U64,
                                                METRIC_VALUE_UNIT_COUNT,
                                                METRIC_VALUE_MONOTONIC,
                                                sinsp_stats_v2->m_n_retrieve_evts_drops));
			m_metrics.emplace_back(new_metric(sinsp_stats_v2_resource_utilization_names[SINSP_STATS_V2_NONCACHED_THREAD_LOOKUPS], \
			                                    METRICS_V2_STATE_COUNTERS,
                                                METRIC_VALUE_TYPE_U64,
                                                METRIC_VALUE_UNIT_COUNT,
                                                METRIC_VALUE_MONOTONIC,
                                                sinsp_stats_v2->m_n_noncached_thread_lookups));
			m_metrics.emplace_back(new_metric(sinsp_stats_v2_resource_utilization_names[SINSP_STATS_V2_CACHED_THREAD_LOOKUPS], \
			                                    METRICS_V2_STATE_COUNTERS,
                                                METRIC_VALUE_TYPE_U64,
                                                METRIC_VALUE_UNIT_COUNT,
                                                METRIC_VALUE_MONOTONIC,
                                                sinsp_stats_v2->m_n_cached_thread_lookups));
			m_metrics.emplace_back(new_metric(sinsp_stats_v2_resource_utilization_names[SINSP_STATS_V2_FAILED_THREAD_LOOKUPS], \
			                                    METRICS_V2_STATE_COUNTERS,
                                                METRIC_VALUE_TYPE_U64,
                                                METRIC_VALUE_UNIT_COUNT,
                                                METRIC_VALUE_MONOTONIC,
                                                sinsp_stats_v2->m_n_failed_thread_lookups));
			m_metrics.emplace_back(new_metric(sinsp_stats_v2_resource_utilization_names[SINSP_STATS_V2_ADDED_THREADS], \
			                                    METRICS_V2_STATE_COUNTERS,
                                                METRIC_VALUE_TYPE_U64,
                                                METRIC_VALUE_UNIT_COUNT,
                                                METRIC_VALUE_MONOTONIC,
                                                sinsp_stats_v2->m_n_added_threads));
			m_metrics.emplace_back(new_metric(sinsp_stats_v2_resource_utilization_names[SINSP_STATS_V2_REMOVED_THREADS], \
			                                    METRICS_V2_STATE_COUNTERS,
                                                METRIC_VALUE_TYPE_U64,
                                                METRIC_VALUE_UNIT_COUNT,
                                                METRIC_VALUE_MONOTONIC,
                                                sinsp_stats_v2->m_n_removed_threads));
			m_metrics.emplace_back(new_metric(sinsp_stats_v2_resource_utilization_names[SINSP_STATS_V2_N_DROPS_FULL_THREADTABLE], \
			                                    METRICS_V2_STATE_COUNTERS,
                                                METRIC_VALUE_TYPE_U32,
                                                METRIC_VALUE_UNIT_COUNT,
                                                METRIC_VALUE_MONOTONIC,
                                                sinsp_stats_v2->m_n_drops_full_threadtable));
			m_metrics.emplace_back(new_metric(sinsp_stats_v2_resource_utilization_names[SINSP_STATS_V2_N_MISSING_CONTAINER_IMAGES], \
			                                    METRICS_V2_STATE_COUNTERS,
                                                METRIC_VALUE_TYPE_U32,
                                                METRIC_VALUE_UNIT_COUNT,
                                                METRIC_VALUE_NON_MONOTONIC_CURRENT,
                                                sinsp_stats_v2->m_n_missing_container_images));
			m_metrics.emplace_back(new_metric(sinsp_stats_v2_resource_utilization_names[SINSP_STATS_V2_N_CONTAINERS], \
			                                    METRICS_V2_STATE_COUNTERS,
                                                METRIC_VALUE_TYPE_U32,
                                                METRIC_VALUE_UNIT_COUNT,
                                                METRIC_VALUE_NON_MONOTONIC_CURRENT,
                                                sinsp_stats_v2->m_n_containers));
		}
```

Reply:
The diff looks rather strange as it is not wide enough to properly show the end result.